src/interpreter/ByteCodeGenerator.cpp

```diff
@@ -210,6 +210,9 @@ ByteCodeBlock* ByteCodeGenerator::generateByteCode(Context* c, InterpretedCodeBl
                     block->pushCode(ReturnFunctionWithValue(ByteCodeLOC(SIZE_MAX), idx), &ctx, nullptr);
                     ctx.giveUpRegister();
                 } else {
+                    if (codeBlock->isGenerator()) {
```

Comment:
express explicitly

Reply:
Could you please explain me what do you mean by that?

Reply:
In order to increase readability, ``if (codeBlock->isGenerator() == true)``

Reply:
Ohh I see that is the part of the new coding convention but wouldn't be a better idea to ensure return value in the getter function instead of checking the returned value after each getter call?

Your suggestion:

```cpp
public:
bool isGenerator()
{
    return m_isGenerator;
}

private:
bool m_isGenerator :1;

// Usage:

if (value.isGenerator() == true) {
 // Yes, at this point you MUST check the returned value, since 
 // the bool type in only a typedef for e.g. unsigned char,
 // so the returned value can be 0-255
 // do stuff
}
```

How about this?

```cpp
#define IS_BOOLEAN(value) ((value == 1) || (value == 0))

public:
bool isGenerator()
{
    ASSERT(IS_BOOLEAN(m_isGenerator));
    return m_isGenerator;
}

private:
bool m_isGenerator :1;

// Usage:

if (value.isGenerator()) {
    // At this point the returned value MUST be 0 or 1
    // do stuff
}
```

Reply:
We've considered about IsXXX style function what you exampled. if ``if (A.isGenerator())`` is allowed, ``if (!A.isGenerator())`` supposed to be allowed, however it turns out degrade readability.
Another example, when A is ``int``, ``if (A)`` is not able to show its semantics properly in most cases whether it means ``if(A != 0)`` or just ``if (A > 0)``.

Reply:
The example for the `int`s is totally acceptable for me as well, but I still can not get on well the boolean getter approach.

> if if (A.isGenerator()) is allowed, if (!A.isGenerator()) supposed to be allowed, however it turns out degrade readability.

According to that the only problem is the `!A.isGenerator()` which degrades the readability. How about applying the same technique like the `operator ==` and the `operator !=`? Since most of the time the `operator !=` returns with `!(operator ==)`.

So what if each boolean member had an `isXXX()` and an `isNotXXX()` getter function? After these you could write `if (A.isNotGenerator())` instead of `if (!A.isGenerator())`.

What do you think about that?

Reply:
Thanks for your suggestion. It has been discussed before also. All members are supposed to have a pair of getter function like not only ``IsXXX`` and but also ``IsNotXXX``, which turns out brings too much redundancy.

Reply:
Okay, thanks for the explanation.